core/eolearn/core/eodata_io.py

```diff
@@ -262,83 +270,69 @@ def load(self):
         with filesystem.openbin(path, "w") as file_handle:
             if compress_level == 0:
-                self._write_to_file(data, file_handle, file_format)
-                return
+                self._write_to_file(data, file_handle)
 
-            with gzip.GzipFile(fileobj=file_handle, compresslevel=compress_level, mode="wb") as gzip_file_handle:
-                self._write_to_file(data, gzip_file_handle, file_format)
+            else:
+                with gzip.GzipFile(fileobj=file_handle, compresslevel=compress_level, mode="wb") as gzip_file_handle:
+                    self._write_to_file(data, gzip_file_handle)
 
-    def _write_to_file(self, data, file, file_format):
+    @abstractmethod
+    def _write_to_file(self, data: _T, file: Union[BinaryIO, gzip.GzipFile]) -> None:
         """Writes data to a file in the appropriate way."""
-        if file_format is MimeType.NPY:
-            return np.save(file, data)
 
-        if file_format is MimeType.GPKG:
-            layer = fs.path.basename(self.path)
-            try:
-                with warnings.catch_warnings():
-                    warnings.filterwarnings(
-                        "ignore",
-                        message="You are attempting to write an empty DataFrame to file*",
-                        category=UserWarning,
-                    )
-                    return data.to_file(file, driver="GPKG", encoding="utf-8", layer=layer, index=False)
-            except ValueError as err:
-                # This workaround is only required for geopandas<0.11.0 and will be removed in the future.
-                if data.empty:
-                    schema = infer_schema(data)
-                    return data.to_file(file, driver="GPKG", encoding="utf-8", layer=layer, schema=schema)
-                raise err
-
-        if file_format in [MimeType.JSON, MimeType.GEOJSON]:
-            if self.feature_type is FeatureType.BBOX:
-                data = data.geojson
-
-            if self.feature_type is FeatureType.TIMESTAMP:
-                data = [timestamp.isoformat() for timestamp in data]
-
-            try:
-                json_data = json.dumps(data, indent=2)
-            except TypeError as exception:
-                raise TypeError(
-                    f"Failed to serialize {self.feature_type} into a JSON file. Make sure that this feature type "
-                    "contains only JSON serializable Python types before attempting to serialize it."
-                ) from exception
-
-            return file.write(json_data.encode())
-
-        raise ValueError(f"Unsupported file format {file_format} for feature type {self.feature_type}.")
-
-    def _decode(self, file, path):
-        """Loads from a file and decodes content."""
-        file_format = MimeType(fs.path.splitext(path)[1].strip("."))
 
-        if file_format is MimeType.NPY:
-            return np.load(file)
+class FeatureIOnumpy(FeatureIO[np.ndarray]):
```

Comment:
Perhaps the name could be `FeatureIONumpy` to better follow naming conventions? (The other `3` classes could also be renamed accordingly.)

Reply:
done, please check if further corrections are desired

Reply:
The only remaining one is `FeatureIOgeodf`. I would rename it to `FeatureIOGeoDf`.

Reply:
I could've sworn i've renamed it. Guess I didnt confirm the renaming... Done now